spec/requests/api/querying_spec.rb

```diff
@@ -101,6 +101,34 @@ def create_vms_by_name(names)
       expect_query_result(:vms, 2, 2)
       expect_result_resources_to_match_hash([{"name" => "redhat_vm"}, {"name" => "vmware_vm"}])
     end
+
+    it 'supports sql friendly virtual attributes' do
+      host_foo =  FactoryGirl.create(:host, :name => 'foo')
+      host_bar =  FactoryGirl.create(:host, :name => 'bar')
+      vm_foo = FactoryGirl.create(:vm, :name => 'vm_foo')
+      vm_bar = FactoryGirl.create(:vm, :name => 'vm_bar')
+      host_foo.vms << vm_foo
+      host_bar.vms << vm_bar
+
+      run_get vms_url, :sort_by => 'host_name', :sort_order => 'desc', :expand => 'resources'
+
+      expect_query_result(:vms, 2, 2)
+      expect_result_resources_to_match_hash([{'name' => 'vm_foo'}, {'name' => 'vm_bar'}])
```

Comment:
From the outside, it looks like this query returned your objects in the order you created them. I think to prove that they were ordered successfully it may be necessary to create three objects out of order, so that you can see that the result put them into some order in a way that couldn't be an accident.